### ISSUE TITLE
eslint-plugin-react(jsx-key) - false positive with missing key inside React.Children.toArray()

### ISSUE THREAD

**Issue by kristho91** [oxc-project/oxc#4421]
This triggers an error with oxlint:

```
{React.Children.toArray(items.map((item) => {
  return (
    <div>
      {item}
    </div>
  );
}))}
```

But it does not with eslint. When using .map inside React.Children.toArray() a key should not be necessary (to my understanding)

**Comment by vHugoObject** [comment]
Can I take this?

**Comment by Dunqing** [comment]
> Can I take this?

Sure

**Comment by AlfredoPrograma** [comment]
Hello folks! Is right to pass a **non children** array of elements to the `React.Children.toArray` function? 

The docs explicitly exposes that 
> `children` parameter should be the value of the `children` prop received by the component.

So I'm not sure keys automatic calculation may work with **non children** array of elements. If anyone has some clue about it, let me know and I contribute to this issue!

**Comment by vHugoObject** [comment]
My apologies, I didn't add an if statement to check for React.Children for fragments. Will open up another pull request.
